cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/operationrunner/InMemorySourceControlOperationRunner.java

```diff
@@ -155,5 +163,45 @@ private Path validateAppConfigRelativePath(RepositoryManager repositoryManager,
     }
     return filePath;
   }
+
+  @Override
+  public ListAppsResponse list(NamespaceId namespace, RepositoryConfig repoConfig) throws ListAppsFailureException,
+    AuthenticationConfigException {
+    try (RepositoryManager repositoryManager = repoManagerFactory.create(namespace, repoConfig)) {
+      String currentCommit = repositoryManager.cloneRemote();
+      Path basePath = repositoryManager.getBasePath();
+
+      FileFilter configFileFilter = getConfigFileFilter();
+      List<File> configFiles = DirUtils.listFiles(basePath.toFile(), configFileFilter);
+
+      List<ListAppResponse> responses = new ArrayList<>();
+      for (File configFile : configFiles) {
+        String applicationName = FileUtils.getNameWithoutExtension(configFile.getName());
+        String fileHash = repositoryManager.getFileHash(basePath.relativize(configFile.toPath()), currentCommit);
+        responses.add(new ListAppResponse(applicationName, fileHash));
+      }
+
+      // Sorting by application name to make response consistent.
+      responses.sort(Comparator.comparing(ListAppResponse::getName));
```

Comment:
Can the above `DirUtils.listFiles` accept list ordering option so that we don't need to do the in-memory sorting again?

Reply:
Sorting files is not very common usecase so IMHO better to keep the lisfiles simple and sort outside when needed.

Reply:
The idea was to make the response order consistent across API calls as lisFiles does not guarantee a order. This would be helpful for in case of pagination if needed. 
I can remove it for now.